lib/std/hash/crc.zig

```diff
@@ -1,285 +1,911 @@
-            const poly = if (algorithm.reflect_input)
-                @bitReverse(@as(I, algorithm.polynomial)) >> (@bitSizeOf(I) - @bitSizeOf(W))
-            else
-                @as(I, algorithm.polynomial) << (@bitSizeOf(I) - @bitSizeOf(W));
-
-            var table: [256]I = undefined;
-            for (&table, 0..) |*e, i| {
-                var crc: I = i;
-                if (algorithm.reflect_input) {
-                    var j: usize = 0;
-                    while (j < 8) : (j += 1) {
-                        crc = (crc >> 1) ^ ((crc & 1) * poly);
-                    }
-                } else {
-                    crc <<= @bitSizeOf(I) - 8;
-                    var j: usize = 0;
-                    while (j < 8) : (j += 1) {
-                        crc = (crc << 1) ^ (((crc >> (@bitSizeOf(I) - 1)) & 1) * poly);
-                    }
-                }
-                e.* = crc;
-            }
-            break :blk table;
-        };
-
-        crc: I,
-
-        pub fn init() Self {
-            const initial = if (algorithm.reflect_input)
-                @bitReverse(@as(I, algorithm.initial)) >> (@bitSizeOf(I) - @bitSizeOf(W))
-            else
-                @as(I, algorithm.initial) << (@bitSizeOf(I) - @bitSizeOf(W));
-            return Self{ .crc = initial };
-        }
-
-        inline fn tableEntry(index: I) I {
-            return lookup_table[@as(u8, @intCast(index & 0xFF))];
-        }
-
-        pub fn update(self: *Self, bytes: []const u8) void {
-            var i: usize = 0;
-            if (@bitSizeOf(I) <= 8) {
-                while (i < bytes.len) : (i += 1) {
-                    self.crc = tableEntry(self.crc ^ bytes[i]);
-                }
-            } else if (algorithm.reflect_input) {
-                while (i < bytes.len) : (i += 1) {
-                    const table_index = self.crc ^ bytes[i];
-                    self.crc = tableEntry(table_index) ^ (self.crc >> 8);
-                }
-            } else {
-                while (i < bytes.len) : (i += 1) {
-                    const table_index = (self.crc >> (@bitSizeOf(I) - 8)) ^ bytes[i];
-                    self.crc = tableEntry(table_index) ^ (self.crc << 8);
-                }
-            }
-        }
-
-        pub fn final(self: Self) W {
-            var c = self.crc;
-            if (algorithm.reflect_input != algorithm.reflect_output) {
-                c = @bitReverse(c);
-            }
-            if (!algorithm.reflect_output) {
-                c >>= @bitSizeOf(I) - @bitSizeOf(W);
-            }
-            return @as(W, @intCast(c ^ algorithm.xor_output));
-        }
-
-        pub fn hash(bytes: []const u8) W {
-            var c = Self.init();
-            c.update(bytes);
-            return c.final();
-        }
-    };
-}
+const impl = @import("crc/impl.zig");
 
-pub const Polynomial = enum(u32) {
-    IEEE = 0xedb88320,
-    Castagnoli = 0x82f63b78,
-    Koopman = 0xeb31d82e,
-    _,
-};
+pub const Crc = impl.Crc;
```

Comment:
https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces

```diff
-std.hash.crc.Crc
+std.hash.Crc
```

Reply:
These will be interface changes beyond the relatively-pure usingnamspace refactor, to be clear.

I'm not sure I can quite do it like that without going back to `usingnamespace` or generating `hash.zig`, I think.  But I could do something like:

```
-std.hash.crc.Crc
+std.hash.crc.Create
```

Reply:
Hmmm I'm mixing up two different concerns here.  I could for that one case, put the interface up at `+std.hash.Crc` manually.  Really, I could move all of the stuff in `std/crc/impl.zig` up to `std/hash.zig` together.